closure/bin/build/jscompiler.py

```diff
@@ -125,7 +131,12 @@ def Compile(compiler_jar_path, source_paths,
   java_version = _ParseJavaVersion(str(_GetJavaVersionString()))
 
   args = _GetJsCompilerArgs(
-      compiler_jar_path, java_version, source_paths, jvm_flags, compiler_flags)
+      compiler_jar_path, java_version, jvm_flags)
+
+  # Write source path arguments to flag file for avoiding "The filename or 
+  # extension is too long" error in big projects.
+  flags_file = _GetFlagFile(source_paths, compiler_flags)
+  args += ['--flagfile', flags_file.name]
 
   logging.info('Compiling with the following command: %s', ' '.join(args))
```

Comment:
It's unfortunate we lose the logging of all the the `--js` flags here. I honestly don't know how much people rely on this output. We can just try it as-is for now and see if anyone complains.